### ISSUE TITLE
Can't pickle classes that define __getattr__

### ISSUE THREAD

**Issue by proteneer** [proteneer/timemachine#335]
in timemachine/lib/potentials.py we define a class that has a custom ```__getattr__```, which breaks pickle

``` python
class InterpolatedPotential(CustomOpWrapper):


    def bind(self, params):
        assert self.get_u_fn().params is None
        self.params = params
        return self


    # pass through so we can use the underlying methods
    def __getattr__(self, attr):
        return getattr(self.args[0], attr)

    def get_u_fn(self):
        return self.args[0]

    def unbound_impl(self, precision):
        return custom_ops.InterpolatedPotential(
            self.get_u_fn().unbound_impl(precision),
            self.args[1],
            self.args[2]
        )

    def bound_impl(self, precision):
        return custom_ops.BoundPotential(
            self.unbound_impl(precision),
            self.params
        )
```

Diagnosis:

https://stackoverflow.com/questions/50156118/recursionerror-maximum-recursion-depth-exceeded-while-calling-a-python-object-w/50158865#50158865

**Comment by proteneer** [comment]
This will be addressed in the unfortunate mega PR #330

**Comment by proteneer** [comment]
resolved
